pandas/tests/plotting/test_hist_method.py

```diff
@@ -293,6 +319,37 @@ def test_hist_column_order_unchanged(self, column, expected):
 
         assert result == expected
 
+    @pytest.mark.parametrize("by", [None, "c"])
+    @pytest.mark.parametrize("column", [None, "b"])
+    @pytest.mark.parametrize("label", [None, "d"])
+    def test_hist_with_legend(self, by, column, label):
+        # GH 6279 - Histogram can have a legend
+        expected_axes_num = 1 if by is None and column is not None else 2
+        expected_layout = (1, expected_axes_num)
+        expected_labels = label or column or ["a", "b"]
+        if by is not None:
+            expected_labels = [expected_labels] * 2
+
+        index = Index(15 * ["1"] + 15 * ["2"], name="c")
+        df = DataFrame(np.random.randn(30, 2), index=index, columns=["a", "b"])
+
+        kwargs = {"legend": True, "by": by, "column": column}
+
+        if label is not None:
```

Comment:
Can you just split this out into another test called `test_hist_with_legend_label_raises`? We typically follow that convention in our test base